miner/src/pool/replace.rs

```diff
@@ -71,6 +71,13 @@ where
 			let old_score = (old.priority(), old.gas_price());
 			let new_score = (new.priority(), new.gas_price());
 			if new_score > old_score {
+				// check for an existing transaction with the same nonce
```

Comment:
```suggestion
				// Check if this is a replacement transaction.
				// 
				// With replacement transactions we can safely return `InsertNew` here, because
				// we don't need to remove `old` (worst transaction in the pool) since `new` will replace
			    // some other transaction in the pool so we will never go above limit anyway.
```